nomt/src/store/mod.rs

```diff
@@ -278,16 +286,28 @@ impl Store {
     ) -> anyhow::Result<()> {
         let mut sync = self.sync.lock();
 
-        sync.sync(
+        if self
+            .shared
+            .poisoned
+            .load(std::sync::atomic::Ordering::Relaxed)
+        {
+            anyhow::bail!("Store is poisoned due to prior error");
+        }
+
+        if let Err(e) = sync.sync(
             &self.shared,
             value_tx,
             self.shared.pages.clone(),
             self.shared.values.clone(),
             self.shared.rollback.clone(),
             page_cache,
             updated_pages,
-        )
-        .unwrap();
+        ) {
+            self.shared
```

Comment:
strictly speaking, it may not be necessary to poison in all these cases. for example, beatree writes don't require poisoning. They just abandon the commit.